classes/omnect_uboot_embedded_version.bbclass

```diff
@@ -0,0 +1,206 @@
+#    the same path but with a suffix '.versioned', the original file gets
+#    suffixed with '.unversioned' and a link gets created from the original
+#    file name to the '.versioned' file so that subsequent processes take the
+#    correct file as input.
+#  - OMNECT_BOOTLOADER_EMBEDDED_VERSION_TYPE
+#    this can be either "file" or "flash", defining of which type the binary is
+#  - OMNECT_BOOTLOADER_EMBEDDED_VERSION_PARAMSIZE
+#    number of bytes to reserve for version information
+#    (and maybe others in future)
+#  - OMNECT_BOOTLOADER_EMBEDDED_VERSION_IMAGESIZE
+#    final size of bootloader binary including version information block.
+#    this parameter is only used for "flash" type bootloaders which have a
+#    pre-defined flash area to reside in, in order to use space at specific
+#    offset from end of region
+#  - OMNECT_BOOTLOADER_EMBEDDED_VERSION_IMAGEOFFSET
+#    only used for type "flash" and only to find correct bootloader location
+#    (and hence version information location) during runtime
+#  - OMNECT_BOOTLOADER_EMBEDDED_VERSION_IMAGEGZ
+#    is set to any value a gzipped version is created, too, during image
+#    creation bootloader.bin.versioned.gz is also created
+#  - OMNECT_BOOTLOADER_EMBEDDED_VERSION_LOCATION
+#    defines the run-time location of the bootloader and is a file path for
+#    type "file"; in case boot loader is part of flash it is assumed that it
+#    resides in the same device as the root filesystem 
+#  - OMNECT_BOOTLOADER_EMBEDDED_VERSION_MAGIC
+#    a magic number precedes the actual version string, followed by a length
+#    byte containing the length of the following string, i.e. w/o magic number
+#    and length byte; it is defined as string containing hexadecimal values
+#    w/o preceding "0x" (as common in many programming languages), e.g.:
+#      "de ad be ef"
+#  - OMNECT_BOOTLOADER_EMBEDDED_VERSION_FILLBYTE
+#    optionally defines the byte to be used to fill space between original
+#    boot loader binary and version data, and also fro end of version data
+#    up to defined size of resulting file; default value is 0x00
+#    however, for NOR flashes it would be better to have an 0xff, therefore
+#    this variable
+#
+# format of version information is:
+#
+#    byteoffset: | 0 ... n | (n + 1)  | (n + 2) ... (n + 1 + *n) |
+#    ------------+---------+----------+--------------------------+
+#    meaning:    | <magic> | <length> | <version string>
+#
+# where:
+#  - <magic> is defined by variable OMNECT_BOOTLOADER_EMBEDDED_VERSION_MAGIC
+#  - <length> is one byte specifying length of <version string>
+#  - <version string> is the actual version, w/o terminating character
+#
+
+python omnect_uboot_embed_version() {
+    import glob
+    import os
+    import shutil
+    import gzip
+    from pathlib import Path
+
+    # read the previously calculated version information
+    bootloader_version_file = d.getVar("WORKDIR") + "/omnect_bootloader_version"
```

Comment:
why not `DEPLOY_DIR_IMAGE`. which mechanism copies it to that path?

Reply:
i'm confused. this is an input, right? "read the previously calculated version information". 
so where does it come from?
`omnect_bootloader_versioning.bbclass` deploys to `DEPLOY_DIR_IMAGE` at parse time.

Reply:
changed to directly using file from DEPLOY_DIR_IMAGE.